econml/_ortho_learner.py

```diff
@@ -0,0 +1,584 @@
+    model_nuisance: estimator
+        The estimator for fitting the nuisance function. Must implement
+        `fit` and `predict` methods that both have signatures:
+
+        .. highlight:: python
+        .. code-block:: python
+
+            model_nuisance.fit(Y, T, X=X, W=W, Z=Z,
+                               sample_weight=sample_weight, sample_var=sample_var)
+            model_nuisance.predict(Y, T, X=X, W=W, Z=Z,
+                                   sample_weight=sample_weight, sample_var=sample_var)
+
+        In fact we allow for the model method signatures to skip any of the keyword arguments
+        as long as the class is always called with the omitted keyword argument set to `None`.
+        This can be enforced in child classes by re-implementing the fit and the various effect
+        methods. If `discrete_treatment=True`, then the input `T` to both above calls will be the
+        one-hot encoding of the original input `T`, excluding the first column of the one-hot.
+
+    model_final: estimator for fitting the response residuals to the features and treatment residuals
+        Must implement `fit` and `predict` methods that must have signatures:
+
+        .. highlight:: python
+        .. code-block:: python
+
+            model_final.fit(Y, T, X=X, W=W, Z=Z, nuisances=nuisances,
+                            sample_weight=sample_weight, sample_var=sample_var)
+            model_final.predict(X=X)
+
+        Predict, should just take the features X and return the constant marginal effect. In fact we allow for the
+        model method signatures to skip any of the keyword arguments as long as the class is always called with the
+        omitted keyword argument set to `None`. Moreover, the predict function of the final model can take no argument
+        if the class is always called with `X=None`. This can be enforced in child classes by re-implementing the fit
+        and the various effect methods. If `discrete_treatment=True`, then the input `T` to both above calls will be
+        the one-hot encoding of the original input `T`, excluding the first column of the one-hot.
+
+    discrete_treatment: bool
+        Whether the treatment values should be treated as categorical, rather than continuous, quantities
+
+    n_splits: int, cross-validation generator or an iterable, optional
+        Determines the cross-validation splitting strategy.
+        Possible inputs for cv are:
+
+        - None, to use the default 3-fold cross-validation,
+        - integer, to specify the number of folds.
+        - :term:`CV splitter`
+        - An iterable yielding (train, test) splits as arrays of indices.
+
+        For integer/None inputs, if the treatment is discrete
+        :class:`~sklearn.model_selection.StratifiedKFold` is used, else,
+        :class:`~sklearn.model_selection.KFold` is used
+        (with a random shuffle in either case).
+
+        Unless an iterable is used, we call `split(X,T)` to generate the splits.
```

Comment:
Now split is called on the concatenation of the non-None arrays of Z, W, and X (if there are any, or a column of ones otherwise).

Reply:
indeed. Fixed docstring accordingly.